src/components/smartSearch/filters/All/index.tsx

```diff
@@ -1,24 +1,58 @@
-import { FormEvent } from 'react';
 import { FormattedMessage as Msg } from 'react-intl';
-import { Box, Button, Divider, Typography } from '@material-ui/core';
-import { FILTER_TYPE, OPERATION, ZetkinSmartSearchFilter } from 'types/smartSearch';
+import { Box, Button, Divider, MenuItem, Typography } from '@material-ui/core';
+import { FormEvent, useState } from 'react';
+
+import StyledSelect from 'components/smartSearch/inputs/StyledSelect';
+import useSmartSearchFilter from 'hooks/useSmartSearchFilter';
+import { DefaultFilterConfig, NewSmartSearchFilter, SmartSearchFilterWithId,
+    ZetkinSmartSearchFilter } from 'types/smartSearch';
 
 interface AllProps {
-    onSubmit: (filter: ZetkinSmartSearchFilter) => void;
+    filter:  SmartSearchFilterWithId<DefaultFilterConfig> | NewSmartSearchFilter;
+    onSubmit: (filter: ZetkinSmartSearchFilter<DefaultFilterConfig> |
+        SmartSearchFilterWithId<DefaultFilterConfig>) => void;
+    onDelete: (filter: SmartSearchFilterWithId<DefaultFilterConfig>) => void;
     onCancel: () => void;
 }
 
-const All = ({ onSubmit, onCancel }: AllProps): JSX.Element => {
+const All = ({ onSubmit, onCancel, onDelete, filter: initialFilter }: AllProps): JSX.Element => {
```

Comment:
I might be wrong here but it feels like treating this as a regular filter like any other makes the code _more_ complicated right now.